### PR TITLE
Linq.js provides its own typings

### LINKED ISSUE #11711: @types/linq is completely outdated
We are in the process of migrating to the new TypeScript 2.0 definitions mechanism. We use `linqjs` in our projects, and we noticed that the corresponding `@types/linq` package contains a very old definitions file, which has several errors, and doesn't even declare a module (so it's not possible to `import` AFAIK).

Please add the latest definitions file as well, `3.0.3-Beta5` for example.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #11711

https://linqjs.codeplex.com/SourceControl/latest#typescript/linq.js.d.ts
I've also emailed the author about this.

### CODE REVIEW

File: linq/linq.3.0.4-Beta5.d.ts [hunk 1]
```diff
@@ -1,276 +0,0 @@
...
-        memoize(): IDisposableEnumerable<T>;
-        catchError(handler: (exception: any) => void): IEnumerable<T>;
-        finallyAction(finallyAction: () => void): IEnumerable<T>;
-        log(): IEnumerable<T>;
-        log<TValue>(selector: (element: T) => TValue): IEnumerable<T>;
-        trace(message?: string): IEnumerable<T>;
-        trace<TValue>(message: string, selector: (element: T) => TValue): IEnumerable<T>;
-    }
-
-    interface IOrderedEnumerable<T> extends IEnumerable<T> {
```
Comment (anubmat) [inline linq/linq.3.0.4-Beta5.d.ts]: Same for all other interfaces

**Comment by anubmat** [comment]
@andy-ms , can you resolve conflicts? and update the PR according to comments..

File: linq/linq.3.0.4-Beta5.d.ts (hunk 1 above)
Comment (andy-ms) [inline linq/linq.3.0.4-Beta5.d.ts]: The package is being removed, so it's not our business any more.

Comment (anubmat) [inline linq/linq.3.0.4-Beta5.d.ts]: Yes, sorry about that!

**Pull request merged**